app/code/community/HE/TwoFactorAuth/controllers/Adminhtml/TwofactorController.php

```diff
@@ -207,13 +219,23 @@ private function _verifyGoogle()
                         );
                     }
 
-                    $msg = Mage::helper('he_twofactorauth')->__("Valid code entered");
-                    Mage::getSingleton('adminhtml/session')->addSuccess($msg);
                     Mage::getSingleton('admin/session')->set2faState(HE_TwoFactorAuth_Model_Validate::TFA_STATE_ACTIVE);
                     $user = Mage::getSingleton('admin/session')->getUser();
                     $password = $user->getPassword();
-                    Mage::dispatchEvent('twofactor_auth_verification_success', array('password' => $password, 'user' => $user, 'result' => true));
-                    $this->_redirect('*');
+                    $lockInfo = new Varien_Object();
+                    Mage::dispatchEvent('twofactor_auth_verification_success', array('password' => $password, 'user' => $user, 'result' => true, 'lock_info' => $lockInfo));
+                    if ($lockInfo->getData('is_locked')) {
+                        Mage::getSingleton('adminhtml/session')->addError(Mage::helper('adminhtml')->__('You did not sign in correctly or your account is temporarily disabled.'));
+                        $this->getRequest()->setParam('forwarded', true)
+                            ->setRouteName('adminhtml')
+                            ->setControllerName('index')
+                            ->setActionName('login')
+                            ->setDispatched(false);
+                    } else {
+                        $msg = Mage::helper('he_twofactorauth')->__("Valid code entered");
```

Comment:
I would want to keep the message

We could create a flag `tfaoff.flag` to bypass the check... assume on standby, I forget my code(or don't have access to a phone) and end up in the admin panel

On the next business day, I would wanna remove the flag and enforce TFA again

I feel this messaging (if missing) could be an indication. 

Happy to remove if not satisfied with my explanation